scripts/screeps_tencent_batch_rl_runner.py

```
@@ -2258,11 +2294,30 @@ def verified_remote_candidate_scorecard(
-        if top_level_runtime_injected or runtime_injected:
+        gradient_blocked = (
+            raw.get("trustedGradientUpdate") is False
+            or raw.get("highVariance") is True
+            or raw.get("missingPrerequisite") == "gradient_stability"
+            or raw.get("classification") == "gradient_stability_untrusted_scorecard_materialized"
+        )
+        if gradient_blocked:
+            if not top_level_runtime_injected or not runtime_injected:
+                raise BatchRunError(
+                    "remote candidateScorecard gradient-stability materialized status requires runtimeParameterInjection proof"
+                )
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)